community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/ReadAheadChannel.java

```diff
@@ -56,7 +56,7 @@ public ReadAheadChannel( T channel, int readAheadSize )
 
     public long position() throws IOException
     {
-        return channel.position();
+        return offset();
```

Comment:
1) Yes
2) Sounds sensible, will update.
3) Will update.
4) I am not sure if it's a bugfix, but it's hard imagining who would like to know how far ahead the underlying channel is instead of the buffered stream on is reading. I will fix and fixate it with a test.